src/charm.py

```diff
@@ -356,13 +356,37 @@ def configure_app_non_leader(self, event: ops.framework.EventBase) -> None:
         if isinstance(event, MicroClusterNodeAddedEvent):
             self.cluster_nodes.join_node_to_cluster(event)
 
+    def _get_bootstrap_params(self) -> dict:
+        """Fetch bootstrap parameters."""
+        micro_ip = cluster_net = public_net = ""
+        try:
+            # Public Network
+            public_nets = self.model.get_binding(binding_key="public").network.interfaces
+            if len(public_nets) > 0:
+                public_net = public_nets[0].subnet
+            # Cluster Network
+            cluster_nets = self.model.get_binding(binding_key="cluster").network.interfaces
+            if len(cluster_nets) > 0:
+                cluster_net = cluster_nets[0].subnet
+            # MicroCeph IP
+            micro_ip = self.model.get_binding(binding_key="admin").network.bind_address
+
+            logger.info(
+                {"public_net": public_net, "cluster_net": cluster_net, "micro_ip": micro_ip}
+            )
+        except ops.model.ModelError as e:
```

Comment:
If this exception is raised, we will run into a further exception down below when trying to access nonexistent variables. We need to reraise the exception to avoid that. Also, imho, I'd use `logger.exception` instead of warning as it's a bit too critical for a warning only.

Reply:
`micro_ip = cluster_net = public_net = ""` We initialise the required variable with empty string value at the very beginning of the method. So if the binding is not available and results in an exception they would fall back to the empty value ?

Reply:
ah, didn't see the initialization at function start. nvm then